### PR TITLE
Add syntax highlighted tooltips

### PR DESCRIPTION
**Pull request by saul**
image

### CODE REVIEW

**Comment by drhumlen** [comment]
Looks really nice, which helps a lot on the readability. 

Would be nice to have this merged asap so I can statt using it 😃

What does it look like when you hover other things? like local on regular functions or a record?

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs [hunk 1]
```diff
@@ -43,61 +107,59 @@ type FSharpIdentifierTooltipProvider(lifetime, solution, presenter, xmlDocServic
...
                       | _ -> () ]
-                    |> String.concat "\n\n"
-                    |> result.Add))
-
-        result.Join(RiderTooltipSeparator)
-
-    override x.GetRichTooltip(highlighter) =
-        RichTextBlock(x.GetTooltip(highlighter))
+                    |> richTextJoin "\n\n"))
+        // todo: add support for _RIDER_HORIZONTAL_LINE_TOOLTIP_SEPARATOR_
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs:161]: @kirillgla Is it supposed to work with the rich text tooltips?

Comment (kirillgla) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs:161]: Yes, it is. I tried it out locally, and I can say that replacing `"\n--------------------\n"` with `"_RIDER_HORIZONTAL_LINE_TOOLTIP_SEPARATOR_"` works just like one would expect it to
image

**Comment by kirillgla** [comment]
Also, I strongly recommend implementing `IRiderQuickDocTooltipRequest` in F#. While syntax highlighting in tooltips might work without it, implementing that interface will make the IDE UI more consistent

**Comment by auduchinok** [comment]
@kirillgla Thanks for the advice! We'll look into migrating to these APIs.

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs (hunk 1 above)
Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs]: Thanks! 🙂

**Review by auduchinok — APPROVED** [review]

**Comment by auduchinok** [comment]
@saul Thanks again for this, it looks really good now!
And props to @kirillgla for the rich text tooltips in the platform. 🙂

**Pull request merged**